packages/node-core/src/indexer/connectionPoolState.manager.ts

```diff
@@ -191,13 +192,13 @@ export class ConnectionPoolStateManager<T extends IApiConnectionSpecific<any, an
   }
 
   //eslint-disable-next-line @typescript-eslint/require-await
-  async setTimeout(endpoint: string, delay: number): Promise<void> {
+  async setRecoverTimeout(endpoint: string, delay: number): Promise<void> {
     // Make sure there is no existing timeout
     await this.clearTimeout(endpoint);
 
     this.pool[endpoint].timeoutId = setTimeout(() => {
       this.pool[endpoint].backoffDelay = 0; // Reset backoff delay only if there are no consecutive errors
-      this.pool[endpoint].rateLimited = false;
+      // this.pool[endpoint].rateLimited = false;   // Do not reset rateLimited status
```

Comment:
Do you think here we need to set a new timeout of the same value so it stayed rate limited?

Reply:
I think adding this here doesn’t avoid the problem; it just delays its appearance.